app/bucketlists/bucketlist.py

```diff
@@ -0,0 +1,239 @@
+import json
+import jwt
+from app import db
+from app.models.bucketlist_models import Bucketlists, Items
+from config import Config
+from datetime import datetime
+from flask import jsonify, request
+from flask_restful import abort, Resource
+
+
+def decode_token(self):
+    """Function to authenticate the token"""
+    token = request.headers.get('Authorization')
+    if not token:
+        abort(401, message="No Authoriztion token")
+    try:
+        payload = jwt.decode(token, Config.SECRET_KEY)
+    except jwt.DecodeError:
+        abort(401, message='Token is invalid')
+    except jwt.ExpiredSignature:
+        abort(401, message='Session expired, please login again')
+
+    return payload['sub']
+
+
+class Bucketlist(Resource):
+    """
+This class contains all the Bucketlists Resource.
+GET: retrieves all the bucketlist a user has.
+?q : Provides a searchparameter
+?limit: Provides a limit of how many items can be displayed on a page
+POST: Adds a bucketlist.
+    """
+
+    def get(self):
+        """Here we list all created bucketlists"""
+        result = {}
+        item_list = []
+        bucketlist_list = []
+        user_id = decode_token(request)
+        query_string = request.args.to_dict()
+        # Here we impliment pagination
+        limit = int(query_string.get('limit', 20))
+        page_no = int(query_string.get('page', 1))
+
+        if type(limit) is not int:
+            abort(400, message="Make sure the limit is an integer")
+        if type(page_no)is not int:
+            abort(400, message="Make sure the limit is an integer")
+        if 'q' in query_string:
+            search_result = Bucketlists.query.filter(
+                Bucketlists.name.ilike(
+                    '%' + query_string['q'] + '%')).filter_by(
+                creator_id=user_id).paginate(
+                page_no, limit)
+
+            if not len(search_result.items):
```

Comment:
` if not len(search_result.items):` can be replaced by  `if not search_result.items:`. Check on `falsy` and `truthy` datatypes [documentation](https://docs.python.org/2.4/lib/truth.html)